lldb/source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -1153,8 +1153,10 @@ SwiftLanguageRuntime::CalculateErrorValue(StackFrameSP frame_sp,
   if (!ast_context || error.Fail())
     return error_valobj_sp;
 
-  lldb::DataBufferSP buffer(new lldb_private::DataBufferHeap(
-      arg0->GetScalar().GetBytes(), arg0->GetScalar().GetByteSize()));
+  auto buffer_up =
+      std::make_unique<DataBufferHeap>(arg0->GetScalar().GetByteSize(), 0);
+  arg0->GetScalar().GetBytes(buffer_up->GetData());
+  lldb::DataBufferSP buffer(std::move(buffer_up));
```

Comment:
I'm just curious: why do we create a unique pointer first and then assign it to a shared pointer? Could we create it as a shared pointer right from the start and then call GetBytes on the raw memory?

Reply:
@adrian-prantl  That's a good question. I was mostly following the example of the line 

https://github.com/llvm/llvm-project/blob/d0fa52cc3797fd8805d24a04e6b8198154cd7b53/lldb/source/Utility/Scalar.cpp#L98

in the patch 

https://github.com/llvm/llvm-project/commit/d0fa52cc3797fd8805d24a04e6b8198154cd7b53